### DISCUSSION TITLE
item vs node - what's the difference

### DISCUSSION THREAD

**Discussion by f00dog-x300** [pytest-dev/pytest#10125] [Q&A] [answered]
Hi all, 

I'm trying to get a better understanding on how some of the pytest hook work. I don't understand what the difference is between using `item` v `node`. What's the use case for either of these?

Thanks in advance. 
Kev

**Comment by RonnyPfannschmidt** [comment, marked as answer]
nodes is anything thats part of the collection tree, (files, modules, classes)

items are the leaves of that tree and the individual tests

**Reply by The-Compiler** [reply]
Since I happen to have a picture of this lying around as part of my pytest trainings already:

![image](https://user-images.githubusercontent.com/625793/178283856-fe0284c6-d58d-4fbd-8a89-b8961e4a0030.png)

**Comment by f00dog-x300** [comment]
Thank @The-Compiler and @RonnyPfannschmidt !
Can any pytest hook use either or are there specific hooks that one can use but not the other? I'm looking at basically trying to perform a REST call if a test passes or if it fails. 

Also I may be misunderstanding something potentially - does one or the other collect the information while the test is running or after all tests are completed?

Edit: The `s` on my keyboard seems to not be working properly today so I had to fix a few typos.
